slides/advanced-03-iterative.qmd

```diff
@@ -0,0 +1,658 @@
+  geom_point(data = bayes_points %>% filter(.iter == 1), aes(y = mean)) +
+  geom_ribbon(aes(ymin = .mean - .sd, ymax = .mean + .sd),
+              alpha = 1 / 7) +
+  scale_x_log10() +
+  labs(y = "MAE", x = "Learning Rate")
+```
+
+
+## Expected Improvement
+
+```{r}
+#| label: gp-iter-1-ei
+#| echo: false
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 6
+#| fig-height: 4.25
+
+scaled_obj %>%
+  filter(.iter == 1) %>% 
+  ggplot(aes(learn_rate, scaled)) +
+  geom_point(data = bayes_points %>% filter(.iter == 1), aes(y = zero)) +
+  geom_line(alpha = 1 / 2, col = "#D95F02", linewidth = 1) +
+  scale_x_log10()  +
+  labs(y = "Expected Improvement", x = "Learning Rate")
+```
+
+
+## GP Evolution
+
+
+```{r}
+#| label: gp-anime
+#| echo: false
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 6
+#| fig-height: 4.25
+
+knitr::include_graphics("animations/anime_gp.gif")
+```
+
+
+## Expected Improvement Evolution
+
+
+```{r}
+#| label: ei-anime
+#| echo: false
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 6
+#| fig-height: 4.25
+
+knitr::include_graphics("animations/anime_improvement.gif")
+```
+
+## BO in tidymodels
+
+We'll use a function called `tune_bayes()` that has very similar syntax to `tune_grid()`. 
+
+<br> 
+
+It has an additional `initial` argument that can take an integer or the previous results of another `tune_*()` function. 
+ 
+ - We'll run the optimization more than once, so let's make an initial grid of results to serve as the substrate for the BO. 
+
+ - I suggest at least the number of tuning parameters plus two as the initial grid for BO. 
+
+## An Initial Grid
+
+```{r tuning} 
+#| label: lgb-bo-initial
+#| cache: true
+reg_metrics <- metric_set(mae, rsq)
+
+set.seed(12)
+init_res <-
+  lgbm_wflow %>%
+  tune_grid(
+    resamples = hotel_rs,
+    grid = nrow(lgbm_param) + 2,
+    param_info = lgbm_param,
+    metrics = reg_metrics
+  )
+
+show_best(init_res, metric = "mae")
+```
+
+## BO using tidymodels
+
+```{r tuning} 
+#| label: lgb-bo
+#| cache: true
+
+set.seed(15)
+lgbm_bayes_res <-
+  lgbm_wflow %>%
+  tune_bayes(
+    resamples = hotel_rs,
+    initial = init_res,     # <- initial results
+    iter = 20,
+    param_info = lgbm_param,
+    metrics = reg_metrics
+  )
+
+show_best(lgbm_bayes_res, metric = "mae")
+```
+
+
+## Plotting BO Results
+
+```{r}
+#| label: autoplot-marginals
+#| echo: true
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 10
+#| fig-height: 4.25
+
+autoplot(lgbm_bayes_res, metric = "mae")
+```
+
+
+## Plotting BO Results
+
+```{r}
+#| label: autoplot-param
+#| echo: true
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 10
+#| fig-height: 4.25
+
+autoplot(lgbm_bayes_res, metric = "mae", type = "parameters")
+```
+
+
+## Plotting BO Results
+
+```{r}
+#| label: autoplot-perf
+#| echo: true
+#| out-width: 50%
+#| fig-align: center
+#| fig-width: 6
+#| fig-height: 4.25
+
+autoplot(lgbm_bayes_res, metric = "mae", type = "performance")
+```
+
+
+## ENCHANCE
```

Comment:
Googled this and it does seem to be a thing, but confirming that this wasn't supposed to be "enhance"?

Reply:
typo